backend/src/services/write/timeBound.ts

```diff
@@ -21,14 +22,19 @@ export const writeTimeBound = async (
   references: Reference[] | undefined,
   authorizer: string
 ) => {
-  const writeHandler = getTimeBoundWriteHandler(timeBound.bid ? 'update' : 'add')
+  const writeHandlerType = timeBound.bid ? 'update' : 'add'
+  const writeHandler = getTimeBoundWriteHandler(writeHandlerType)
 
   try {
     await writeHandler.start()
     const result = await writeHandler.upsertObject('now_tu_bound', timeBound, ['bid'])
-    writeHandler.idValue = result!.bid as number
+    if (result) {
+      writeHandler.idValue = result.bid as number // this happens only if a new timebound is created
+    } else {
+      writeHandler.idValue = timeBound.bid as number
+    }
```

Comment:
This could be `writeHandler.idValue = result ? result.bid as number : timeBound.bid as number;`